include/envoy/network/connection_handler.h

```diff
@@ -37,12 +37,13 @@ class ConnectionHandler {
                               const Network::ListenerOptions& listener_options) PURE;
 
   /**
-   * Find a listener based on the provided listener port value.
-   * @param port supplies the port value
+   * Find a listener based on the provided listener address value.
+   * @param address supplies the address value.
    * @return a pointer to the listener or nullptr if not found.
    * Ownership of the listener is NOT transferred
    */
-  virtual Network::Listener* findListenerByPort(uint32_t port) PURE;
+  virtual Network::Listener*
+  findListenerByAddress(const Network::Address::InstancePtr& address) PURE;
```

Comment:
That doesn't make sense, as you end up comparing them in the end. https://github.com/lyft/envoy/blob/master/include/envoy/network/address.h#L80

You had some other compiler error. Please try again or ask someone for an over the shoulder check.